src/components/SidePanes/CatalogInclusionPane.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Form, Spinner } from '@edx/paragon';
+import { PUBLISHED, UNPUBLISHED } from '../../data/constants';
+
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+import Pane from './Pane';
+
+const CatalogInclusionPane = ({ courseUuid, subInclusion, draftStatuses }) => {
+  const [inclusion, setInclusion] = useState(subInclusion);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const message = 'Self-paced courses from participating partner organizations can be included in the Subscription Catalog.';
+  const errorMessage = 'We were unable to toggle this attribute. Please try again later.';
+  const stateError = 'Edits are not allowed while all course runs are in review.';
+
+  const toggleInclusion = async (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+    let draft = null;
+    if (draftStatuses.includes(PUBLISHED)) {
+      draft = false;
+    } else if (draftStatuses.includes(UNPUBLISHED)) {
+      draft = true;
+    } else {
+      // if it hits this condition, the only course runs are in review, so we will block
+      // edits at the course level
+      setIsLoading(false);
+      setError(stateError);
+    }
+
+    if (draft !== null) {
+      const data = {
+        uuid: courseUuid,
+        draft,
+        enterprise_subscription_inclusion: !(inclusion),
+      };
+      try {
+        const response = await DiscoveryDataApiService.editCourse(data);
+        setInclusion(response.data.enterprise_subscription_inclusion);
+        setError(null);
+        setIsLoading(false);
+      } catch (err) {
+        setIsLoading(false);
+        setError(errorMessage);
```

Comment:
Re-iterating, showing the error from caught exception would be more helpful than showing a default error.